### DISCUSSION TITLE
通过简悦自动备份文章到 DEVONthink

### DISCUSSION THREAD

**Discussion by Kenshin** [Kenshin/simpread#3235] [服务]
> 得益于简悦的开放性与基于本地系统，可以跟很多不同的生产力工具联动，以下是 [简悦社区](https://t.me/simpreadgroup) 用户 [FtgsgG](https://www.zhihu.com/people/xu-hao-84-8) 带来了一个教程，原文 [请看这里](https://zhuanlan.zhihu.com/p/449043131)。

https://user-images.githubusercontent.com/81074/147346117-027770e1-266e-4fea-be2e-fea90fd292df.mp4

平时我有将喜欢的文章离线保存到Devonthink的习惯，以便后期检索或编辑。我选择保存为Textbundle以便图片和Markdown分离。

## 在DT中新建数据库

![](https://user-images.githubusercontent.com/81074/147346219-fd1910bd-2765-4819-8846-fb853b83e1ba.png)

![](https://user-images.githubusercontent.com/81074/147346238-45f446bd-b049-4277-8f73-b3f0a8916db6.png)

## 为新建的数据库分组

右键新建的数据库，添加分组。

![](https://user-images.githubusercontent.com/81074/147346406-92798f35-c1a2-48bf-914c-059c89ea4dcd.png)

![](https://user-images.githubusercontent.com/81074/147346423-c1fa7c96-04ff-4ba7-8134-3134847532c8.png)

## 在简悦·同步助手下设置导出的路径

先在本地创建一个Devonthink的目录，如果你想多端同步的话，可以考虑放在iCloud drive下或者Dropbox的同步目录下。

![](https://user-images.githubusercontent.com/81074/147317050-3a0864a3-8fcc-4186-9abd-8f657237b24c.png)

## 使用简悦导出为Textbundle格式

![](https://user-images.githubusercontent.com/81074/147317097-08c8eb19-bcf1-4a95-a9ef-80fa173f251d.png)

圆盘为简悦的Assistive Touch插件，或者：

![](https://user-images.githubusercontent.com/81074/147346766-91b3fca8-c922-49a3-8d63-1b7a5cf9be39.png)

## 设置Hazel

![](https://user-images.githubusercontent.com/81074/147317068-7b4d2025-c56e-4778-95aa-675019677001.png)

这里设置文件创建时间在10分钟之前的文件才会激活脚本，因为简悦导出文章需要时间，而且并非一次性将整个Textbundle保存在目录下，如果你是保存为PDF格式就不会出现这个问题了。

在下面我们选择 embedded script，并在后面的Edit script编写脚本。

```
set DB to "test" -- 这是要使用的数据库的名称
set dest to "/test1" -- 这是要使用的组。必须以这种方式输入组位置的路径

tell application id "DNtp"
	if DB is not in (name of databases) then -- 您需要检查数据库是否已打开
		display alert "Database " & DB & " is not open!" & return & return & "Please open it."
		return
	else
		set destinationGroup to get record at dest in database DB -- 您需要指定数据库，否则将查询当前数据库。这很可能是因为你在工作，而Hazel在后台工作
		set newRecord to (import theFile to destinationGroup)
		if (exists newRecord) then tell application "Finder" to delete theFile
	end if
end tell
```

这样在你点击简悦的导出之后，Hazel就会自动帮你备份到Devonthink中。

`if (exists newRecord) then tell application "Finder" to delete theFile`

这一行是在帮你完成备份之后自动删除你本地的文件，如果你不需要的话请删除这一行。

## 效果

![](https://user-images.githubusercontent.com/81074/147317087-4636018d-879d-4552-b610-00ec6adc64a5.png)

## 参考

https://discourse.devontechnologies.com/t/devonthink-hazel-and-automator/21749/17

**Comment by nickcheng** [comment]
现在似乎用不到 Hazel 了, 直接指定保存位置就好了.

**Reply by Kenshin** [reply]
是的，一直都可以这样。

PS：如果不设定到同步盘的 `output` 而指向 DEVONthink 目录的话，则无法使用 资料库等开放平台的功能。（当然，不是每个用户都需要这些。

**Comment by xooooooooox** [comment]
dt3，有个功能，全局inbox，类似于zotero，dt3会监控这个目录，任何丢到计算机的这个inbox目录中的文件，都会被自动import到dt3中

hazel 这玩意儿，我也买了，学艺不精，感觉没用到精髓。

**Reply by Kenshin** [reply]
> hazel 这玩意儿，我也买了，学艺不精，感觉没用到精髓。

这个不错，能有个简单的教程说明吗？dt3 我只用它来索引，然后检索...

> hazel 这玩意儿，我也买了，学艺不精，感觉没用到精髓。

要是会写点 bash / js 会挺好用的，简悦有不少跟 Hazel 联动方案，你可以搜搜，下周我也会来个跟 Hazel 更联动的玩法。

**Reply by xooooooooox** [reply]
@Kenshin 
很简单的，下午我给你截个图，你就明白了，相当方便。

折腾homelab、内网穿透....都快胃出血了，

**Reply by xooooooooox** [reply]
> > hazel 这玩意儿，我也买了，学艺不精，感觉没用到精髓。
> 
> 这个不错，能有个简单的教程说明吗？dt3 我只用它来索引，然后检索...
> 
> > hazel 这玩意儿，我也买了，学艺不精，感觉没用到精髓。
> 
> 要是会写点 bash / js 会挺好用的，简悦有不少跟 Hazel 联动方案，你可以搜搜，下周我也会来个跟 Hazel 更联动的玩法。

教程来了

![CleanShot 2022-02-12 at 14 14 19@2x](https://user-images.githubusercontent.com/21237993/153699802-91cd23af-3813-4c3e-b530-2916a6a14fc6.png)

**Reply by Kenshin** [reply]
okay 收到~
